src/Archive/VolFile.cpp

```diff
@@ -283,6 +283,11 @@ namespace Archive
 		// Calculate the zero padded length of the string table and index table
 		volInfo.paddedStringTableLength = (volInfo.stringTableLength + 7) & ~3;
 		volInfo.paddedIndexTableLength = (volInfo.indexTableLength + 3) & ~3;
+
+		if (volInfo.indexEntries.size() == 0) {
+			return;
+		}
```

Comment:
No objections here. This is likely how I would have done it. Efficient, and avoids excessive indentation. Just wanted to comment that I needed to more closely examine the end of this routine to make sure the early exit was appropriate. It also occurred to me that if code is later added to the end of the routine, some special consideration may need to be made for the early exit here.